### PR TITLE
Gem rake cleanup badges etc

### PR DESCRIPTION
**Pull request by ndushay**
- group gems better
- capistrano: don't deploy unnec gem group
- capistrano: get rid of non-existent VM  deploy
- add badges
- add Code Climate test reporting
- add rubocop to default ci task

closes sul-dlss/media#62

### CODE REVIEW

**Comment by jmartin-sul** [comment]
@ndushay:  i added some commits... when you get a chance, could you look over them and see whether you have any objections to what i've changed?  i arrived at the changes by testing and fixing what didn't work.  since the commit history was already a bit messy, and since i wanted to make it easy to see what i changed compared to your original version of the branch, i didn't squash anything.  i can squash stuff if you're ok with the changes and you'd prefer a cleaner history.

my testing was as follows:
* on my laptop, clone the argo repo, check out this branch
 * run through the setup instructions in the README for a clean installation (`rake argo:install` etc)
 * load fixtures into dev and test environments (for fedora and solr) as per the readme instructions
 * run `rspec`
 * stop the running jetty instance, run `RAILS_ENV=test bundle exec rake ci`
 * reload the `development` fixtures (since CI does a clean install of jetty)
 * run `rake rubocop` (wasn't looking for clean results, just that it ran, which it did)
 * test the following rake tasks: `about`, `about`, `argo:bump_version`, `argo:htaccess`, `argo:install`, `argo:reindex_all`, `argo:repo:load`, `argo:workgroups`, `integration_tests`, `jetty:start`, `jetty:stop`, `jetty:status`, `rake middleware`, `rake routes`, `stats`, `sul_styles:check_local_colors`
* deploy to stage
 * confirm deployment worked
 * confirm pry is the rails console shell
 * run `rake argo:reindex_all`, confirm that reindexing proceeds as expected (by looking at the logs, and at delayed job info from the rails console; kill early by way of `rake job:clear`, since a full reindex isn't actually the functionality being tested).

and CI still seems good on travis.

File: Gemfile [hunk 1]
```diff
@@ -42,37 +42,40 @@ gem 'blacklight', '~> 5.18'
...
 gem 'sul_styles', '~> 0.3'
 
+# useful for debugging, even in prod
+gem 'pry-byebug' # Adds step-by-step debugging and stack navigation capabilities to pry using byebug
+gem 'pry-rails' # use pry as the rails console shell instead of IRB
+
 group :test, :development do
-  gem 'http_logger'
-  gem 'rspec-rails', '~> 3'
   gem 'capybara'
```
Comment (ndushay) [inline Gemfile]: maybe include a line note that capybara is used by wfs_rails  in development?

**Comment by ndushay** [comment]
Great work, John.  Thanks for the mop up.   I'm okay leaving the commit history as-is.  No strong opposition to squashing it either.  So this is now 👍 from me, but since I did part of it, I won't be the one to merge it.

**Comment by ndushay** [comment]
I would like to propose we squash commits about Gemfile reorg together.   I think all of these can be a single commit:  reorged Gemfile blah blah.

 	af06f8d
 	57ba622
 	4daf358
 		9124de1
2e0764b
 	ec142f0
 	870e746

I would also squash the rubocop thing about single quotes into whatever else makes sense.

But if the rebase doesn't work smoothly, I would settle for a subset of the above that does work smoothly.

**Pull request merged**